org.eclipse.xtext/src/org/eclipse/xtext/formatting2/internal/HiddenRegionReplacer.java

```diff
@@ -56,26 +68,52 @@ protected void applyHiddenRegionFormatting(List<ITextReplacer> replacers) {
 		} else if (dec != null) {
 			separator.setIndentationDecrease(dec);
 		}
+	}
 
-		// 2. apply NewLine and space configuration to the first whitespace region that follows or contains a linewrap.  
+	/**
+	 * Apply NewLine and space configuration to the first whitespace region that follows or contains a linewrap.
+	 * @since 2.15 
+	 */
+	protected void applySeparatorConfiguration(IHiddenRegionFormatting separator) {
 		separator.setNewLinesDefault(formatting.getNewLineDefault());
 		separator.setNewLinesMax(formatting.getNewLineMax());
 		separator.setNewLinesMin(formatting.getNewLineMin());
 		separator.setSpace(formatting.getSpace());
+	}
 
-		// 3. apply the 'priority' configuration for all and set a default formatting
-		for (ITextReplacer replacer : replacers)
+	/**
+	 * @since 2.15
+	 */
+	protected void applyPriorityAndDefaultFormatting(List<ITextReplacer> replacers, IHiddenRegionFormatting separator) {
+		for (int i = 0; i < replacers.size(); i++) {
+			ITextReplacer replacer = replacers.get(i);
 			if (replacer instanceof WhitespaceReplacer) {
 				IHiddenRegionFormatting formatting2 = ((WhitespaceReplacer) replacer).getFormatting();
 				formatting2.setPriority(formatting.getPriority());
 				if (formatting2 != separator) {
```

Comment:
Is it correct to use identity comparison here?

Reply:
Yes, but you could also leave it out or use `equals`. The `if`'s just a very small optimalisation here